src/modules/hg_branch.rs

```diff
@@ -74,26 +73,36 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-fn get_hg_branch_name(ctx: &Context) -> String {
-    std::fs::read_to_string(ctx.current_dir.join(".hg").join("branch"))
-        .map_or_else(|_| "default".to_string(), |s| s.trim().into())
+fn get_hg_repo_root<'a>(ctx: &'a Context) -> Result<&'a Path, Error> {
+    let dir = ctx.current_dir.as_path();
+    let dev_id = dir.device_id();
```

Comment:
You will need to sprinkle `#[cfg(unix)]` here or add a no-op for other targets.

Reply:
It won't compile under windows if `device_id` is not defined in the Path extension, so I guess a stub is the best way to go.